packages/engine/src/avatar/functions/avatarFunctions.ts

```diff
@@ -458,15 +460,36 @@ export function addBoneOpacityParamsToMaterial(material, boneIndexes: Matrix4) {
 
 export const setAvatarHeadOpacity = (entity: Entity, opacity: number): void => {
   const object3DComponent = getComponent(entity, Object3DComponent)
+  let pastOpacity = 1
   object3DComponent?.value.traverse((obj) => {
     if (!(obj as SkinnedMesh).isSkinnedMesh) return
     const material = (obj as SkinnedMesh).material as Material
     if (!material.userData?.shader) return
     const shader = material.userData.shader
     if (shader?.uniforms) {
+      pastOpacity = shader.uniforms.boneOpacity.value
       shader.uniforms.boneOpacity.value = opacity
     }
   })
+
+  const headBone = findHeadBone(object3DComponent?.value)
+  if (!headBone || pastOpacity == opacity) return
+  if (opacity == 0) {
+    ;(headBone as any).traverse((o) => {
+      o.position.set(NaN, NaN, NaN)
+      o.matrixWorld.set(NaN, NaN, NaN, NaN, NaN, NaN, NaN, NaN, NaN, NaN, NaN, NaN, NaN, NaN, NaN, NaN)
```

Comment:
we should not introduce NaN if possible, it will cause crashes if it gets into netcode or physics simulation. is there another way we can do this?

Reply:
Yes. I got it. I will find another way.